Fix: storing IPFS folder CID in OG Founders NFT contract

cybervoid0 opened the pull request:
This pull request fixes the OG Founders NFT contract to store the IPFS folder CID in the contract. It adds a new event `IpfsFolderChanged` to track changes in the IPFS folder. The `initialize` function is updated to include the `maxSupply` and `ipfsFolderCid` parameters. The `setIpfsFolder` function is added to allow an admin to update the IPFS folder and maximum supply of tokens.

cybervoid0 committed:
fix(OG Founders NFT): store IPFS folder CID in the contract

cybervoid0 committed:
doc: add OGFoundersRootstockCollective implementation and proxy addresses

TravellerOnTheRun commented on contracts/NFT/OGFoundersRootstockCollective.sol:
```diff
@@ -24,19 +25,37 @@ contract OGFoundersRootstockCollective is ERC721NonTransferrableUpgradable {
   uint256 public firstProposalDate;
   // Counter for the total number of minted tokens
-  uint8 private _totalMinted;
+  uint256 private _totalMinted;
```
Why have you changed for uint256? What's the purpose? It will never get past 256, cause the total is 150 and for this collection that's it.

jessgusclark replied:
I was under the impression that if you use uint8 in isolation the compiler will automatically convert it to uint256. The way to prevent this would be to have enough uint8s to cover the entire slot. That being said, not the time to debate this.

cybervoid0 replied:
Because now it’s possible to adjust the token quantity using the setIpfFOlder function. The gas savings here are minimal and matter far less than the potential overflow that could occur when adding new tokens, where the uint8 array may not be large enough to accommodate everything.

TravellerOnTheRun commented on ignition/modules/OGFoundersModule.ts:
```diff
@@ -8,13 +8,20 @@ export const OGFoundersModule = buildModule('OGFounders', m => {
+    m.encodeFunctionCall(
+      implementation,
+      'initialize',
+      [deployer, stRIFAddress, firstProposalDate, maxSupply, ipfsFolderCid],
```
Why are you setting maxSupply from the deployment? What's the purpose of this change?

cybervoid0 replied:
Because now the token quantity depends on the number of JSON files in the IPFS folder, and it should be set dynamically either in the initializer or in the setIpfsFolder function.

shenshin replied:
Can you guarantee that tomorrow Toby wont come and ask to replace some images or to add some? It already happened in the past and will happen again. So it must be possible to change te number of tokens in the contract

TravellerOnTheRun reviewed (dismissed):
Unclear changes made which to me look absolutely unnecessary considering the fact that each collection will have its own contract and this one will not have more than 150 tokens

cybervoid0 commented:
Can you guarantee that tomorrow Toby wont come and ask to replace some images or to add some? It already happened in the past and will happen again. So it must be possible to change te number of tokens in the contract

cybervoid0 committed:
fix(OG Founders NFT): update setIpfsFolder function

This commit updates the setIpfsFolder function in the OGFoundersRootstockCollective contract. It adds a requirement that the newMaxSupply must be greater than or equal to the current max supply. This ensures that the new max supply is valid.

jessgusclark reviewed (approved):
Thank you for the quick work!
